### PR TITLE
Adding missing includes which are necessary for building.

### PR DESCRIPTION
**Pull request by mfixman**
The current version of Faiss cannot be built due to missing `#include`s on a test file `tests/test_ivf_index.cpp`. This is better described in issue [#3532](https://github.com/facebookresearch/faiss/issues/3532).

This adds the missing imports.

### CODE REVIEW

File: tests/test_ivf_index.cpp [hunk 1]
```diff
@@ -12,6 +12,8 @@
 #include <cstring>
 #include <map>
 #include <random>
+#include <set>
+#include <algorithm>
```
Comment (junjieqi) [inline tests/test_ivf_index.cpp:16]: How about removing algorithm? I don't see where we use it

Comment (mfixman) [reply tests/test_ivf_index.cpp:16]: It's used in a `std::find` call in this file.

Comment (junjieqi) [reply tests/test_ivf_index.cpp:16]: Hi @mfixman, I think you are right. I'm wondering if you could put #include<alogrithm> in between #include <omp.h> and #include <cstddef>. The format doesn't pass. Once you fix it, I could merge it

**Comment by junjieqi** [comment]
fixed

**Pull request closed without merging**